docs/operating-eck/webhook.asciidoc

```diff
@@ -1,131 +1,191 @@
-Like the ValidatingWebhookConfiguration, it must be created before starting the operator, even if it is empty. By default its name is `elastic-webhook-server-cert`.
-The content of this Secret and the lifecycle of the certificates are automatically managed for you. ECK generates a dedicated and separate certificate authority and ensures that all components are rotated before the expiration date. The certificate authority is also used to configure the `caBundle` field of the `ValidatingWebhookConfiguration`. You can disable this feature if you want to manage the certificates yourself or with https://github.com/jetstack/cert-manager[cert-manager]. See <<{p}-webhook-cert-manager,this example>>.
+* Validate all known Elastic custom resources ({eck_resources_list}) on create and update.
+* The operator itself is the webhook server -- which is exposed through a service named `elastic-webhook-server` in the `elastic-system` namespace.
+* The operator generates a certificate for the webhook and stores it in a secret named `elastic-webhook-server-cert` in the `elastic-system` namespace. This certificate is automatically rotated by the operator when it is due to expire.
 
-[id="{p}-webhook-cert-manager"]
-== Manage the webhook certificate with cert-manager
 
-If ECK is currently running, first make sure that the automatic certificate management feature is disabled. To do this, add the `--manage-webhook-certs=false` flag to the operator deployment manifest.
+[float]
+[id="{p}-{page_id}-manual-config"]
+== Manual configuration
+
+If you installed ECK without the webhook and want to enable it later on, or if you want to customise the configuration such as providing your own certificates, this section describes the options available to you.
+
+[float]
+[id="{p}-{page_id}-config-options"]
+=== Configuration options
+
+You can customise almost all aspects of the webhook setup by changing the <<{p}-operator-config,operator configuration>>.
 
-Then, install cert-manager v0.11+ as described in the https://docs.cert-manager.io/en/latest/getting-started/install/[cert-manager documentation].
+[width="100%",cols=".^10m,.^30d,.^60d",options="header",grid=rows]
+|===
+|Configuration option | Default value |Description
+|enable-webhook       | false   | This must be set to `true` to enable the webhook server.
+|manage-webhook-certs | true    | Set to `false` to disable auto-generating the certificate for the webhook. If disabled, you must provide your own certificates using one of the methods described later in this document.
+|webhook-cert-dir     | /tmp/k8s-webhook-server/serving-certs | Path to mount the certificate
+|webhook-name         | elastic-webhook.k8s.elastic.co | Name of the `ValidatingWebhookConfiguration`. As webhook configurations are cluster resources (not restricted to a single namespace), this allows you to run multiple copies of the operator with each instance having its own webhook.
```

Comment:
I don't think this is fully correct. I a user were to do that would not all registered webhooks of all ECK instances be called for each submitted manifest?

I opened https://github.com/elastic/cloud-on-k8s/issues/3431 a while ago to improve this.

Reply:
Good point!